File: src/main/java/no/novari/integration/model/dtos/IntegrationPostDto.java
package no.novari.integration.model.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

@Getter
@EqualsAndHashCode
@Jacksonized
@Builder
public class IntegrationPostDto {

    @NotNull
    private Long sourceApplicationId;

    @NotBlank
    private String sourceApplicationIntegrationId;

    @NotBlank
    private String destination;

}

File: src/main/kotlin/no/novari/flyt/integration/api/dto/IntegrationPostDto.kt
package no.novari.flyt.integration.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

@Schema(description = "Request to create a Flyt integration.")
data class IntegrationPostDto(
    @field:NotNull
    val sourceApplicationId: Long? = null,
    @field:NotBlank
    val sourceApplicationIntegrationId: String? = null,
    @field:NotBlank
    val destination: String? = null,
)
